### PR TITLE
disttask: add proto for load data

### LINKED ISSUE #42452: disttask: implement proto for load data

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #42452 

## NOTE
This defines only the smallest proto to achieve load data, and further things will be added to it subsequently. You can see the concrete implementation in [this demo](https://github.com/GMHDBJD/tidb/tree/addLoadDataProto/disttask/loaddata)

Problem Summary:

### What is changed and how it works?
- add proto for load data

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/proto/task.go [hunk 1]
```diff
@@ -96,13 +96,17 @@ type MinimalTask interface {
 const (
 	// TaskTypeExample is TaskType of Example.
 	TaskTypeExample = "Example"
+	// LoadData is TaskType of LoadData.
+	LoadData = "LoadData"
 )
 
 // Type2Int converts task type to int.
 func Type2Int(t string) int {
```
Comment (lance6716) [inline disttask/framework/proto/task.go:104]: it's more readable that we assign `TaskTypeExample` and `LoadData` a distinguish type, and associate these converting functions with the type

File: disttask/loaddata/proto.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/parser/mysql"
+)
+
+// TaskStep of LoadData.
+const (
+	ReadSortImport = 1
+)
+
+// Task is the task of LoadData.
```
Comment (lance6716) [inline disttask/loaddata/proto.go:30]: Can you add more comment about the difference of Task/Subtask/MinimalTask? For example, they need to be persisted to somewhere, what's the key concept that they has a 1:1 relation...

File: disttask/loaddata/proto.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	Dir    string
+	Chunk  Chunk
+	Writer *backend.LocalEngineWriter
+}
+
+// IsMinimalTask implements the MinimalTask interface.
+func (MinimalTask) IsMinimalTask() {}
+
+// Table records the table information.
+type Table struct {
```
Comment (lance6716) [inline disttask/loaddata/proto.go:59]: In future I hope they are replaced by the load data plan
https://github.com/pingcap/tidb/blob/3c942f7e101d6b0aff85c552598678f2a761deca/planner/core/common_plans.go#L547-L565

**Review by lance6716 — APPROVED** [review]

File: disttask/loaddata/proto.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	"github.com/pingcap/tidb/parser/mysql"
+)
+
+// TaskStep of LoadData.
+const (
+	ReadSortImport = 1
+)
+
+// Task is the task of LoadData.
+type Task struct {
```
Comment (zimulala) [inline disttask/loaddata/proto.go:31]: `TaskMeta` is better? Otherwise, it will get confused with `proto.Task`.

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**